src/app/public/modules/ag-grid/ag-grid-data-manager-adapter.directive.ts

```diff
@@ -0,0 +1,207 @@
+import {
+  AfterContentInit,
+  ChangeDetectorRef,
+  ContentChildren,
+  Directive,
+  Input,
+  OnDestroy,
+  QueryList
+} from '@angular/core';
+
+import {
+  Subject,
+  Subscription
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  AgGridAngular
+} from 'ag-grid-angular';
+
+import {
+  ColumnMovedEvent,
+  RowSelectedEvent
+} from 'ag-grid-community';
+
+import {
+  SkyDataManagerService,
+  SkyDataManagerSortOption,
+  SkyDataManagerState,
+  SkyDataViewConfig
+} from '@skyux/data-manager';
+
+@Directive({
+  selector: '[skyAgGridDataManagerAdapter]'
+})
+export class SkyAgGridDataManagerAdapterDirective implements AfterContentInit, OnDestroy {
+
+  @Input()
+  private viewId: string;
+
+  @ContentChildren(AgGridAngular, { descendants: true })
+  public agGridList: QueryList<AgGridAngular>;
+
+  private currentAgGrid: AgGridAngular;
+  private dataStateSub: Subscription;
+  private viewConfig: SkyDataViewConfig;
+  private _ngUnsubscribe = new Subject();
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private dataManagerSvc: SkyDataManagerService) { }
+
+  public ngAfterContentInit(): void {
+    this.viewConfig = this.dataManagerSvc.getViewById(this.viewId);
+    this.checkForAgGrid();
+
+    this.agGridList.changes
+    .pipe(takeUntil(this._ngUnsubscribe))
+    .subscribe(() => this.checkForAgGrid());
+  }
+
+  public ngOnDestroy() {
+    this._ngUnsubscribe.next();
+    this._ngUnsubscribe.complete();
+  }
+
+  private checkForAgGrid(): void {
+    const agGridCount = this.agGridList.length;
+
+    /* istanbul ignore else */
+    if (agGridCount > 1) {
+      console.error(
+        'A data view may only have one active ag-Grid child component instance at a time.'
```

Comment:
What if instead of logging a console error and not registering the grid, we register the first grid and log a warning like "More than one ag-Grid child component was found. Using the first ag-Grid."?